macros/src/lib.rs

```diff
@@ -0,0 +1,99 @@
+extern crate proc_macro;
+extern crate core;
+
+use core::panic;
+use proc_macro::TokenStream;
+use std::str::FromStr;
+use quote::quote;
+use solana_sdk::pubkey::{ParsePubkeyError, Pubkey};
+use syn::{parse_macro_input, LitStr};
+
+const TARGET_KEY_LENGTH: usize = 44;
+const MIN_IDENTIFIER_LENGTH: usize = 12;
+const FAKE_ADDRESS_PAD_CHAR: char = '2';
+
+/// Creates a fake base58 public key via the solana_sdk::pubkey! proc macro, padding
+/// the passed in string literal with 2's up to a 44 char address.
+///
+/// Due to some chars not being included in the base58 encoding, the following chars are
+/// converted to their counterpart.
+/// ```text
+/// 'I' => 'i'
+/// 'O' => 'o'
+/// 'l' => 'L'
+/// '0' => 'o'
+/// ```
+///
+/// The last 12 chars are reserved for the identifier, in a case where the resulting
+/// byte array from a pubkey of 44 chars exceeds 32 bytes. The macro will reduce the
+/// length of the reserved space until the correct key size is created.
+///
+/// A valid base58 address is anywhere between 32 and 44 chars specified on the
+/// [Solana CLI Docs](https://docs.solana.com/cli/transfer-tokens). The minimum length
+/// pubkey that will be produced is 32.
+#[proc_macro]
+pub fn named_pubkey(input: TokenStream) -> TokenStream {
+    let input_str = parse_macro_input!(input as LitStr);
+    let input_value = input_str.value();
+
+    let pad_length = TARGET_KEY_LENGTH - input_value.len();
+    if pad_length < MIN_IDENTIFIER_LENGTH {
+        panic!("Input address literal and minimum identifier length is greater than target key");
+    }
+
+    // Generate the fake public key string
+    let fake_pubkey_str = get_named_base58_str(input_value, pad_length);
+
+    let expanded = quote! {
+        solana_sdk::pubkey!(#fake_pubkey_str)
+    };
+    TokenStream::from(expanded)
+}
+
+fn get_named_base58_str(identifier: String, max_pad_length: usize) -> String {
```

Comment:
Can we make this function return a `Result<String, ParsePubkeyError>` instead of panic, and then panic in the macro?

Reply:
And this function is so useful on its own for programmatic naming, I wonder if we could expose it as public, and maybe move it to a different crate? Not sure which one, though. Your call.

Reply:
Alright, made it actually just return the pubkey because we create it anyways and most of the time we will need it. 

I moved it into anchor-utils although I don't really know if that's the correct spot for it? I'll leave that up to you actually!